src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,94 @@
+package storage;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import exception.JarvisException;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.TaskList;
+import task.Todo;
+
+public class Storage {
+    private final String dataFilePath;
+
+    /**
+     * Constructs a Storage object.
+     *
+     * @param dataFilePath Path of the file to store and retrieve data from.
+     */
+    public Storage(String dataFilePath) {
+        this.dataFilePath = dataFilePath;
+    }
+
+    /**
+     * Loads data from the file path.
+     *
+     * @return List of tasks stored in the file.
+     * @throws JarvisException If file retrieval failed.
+     */
+    public List<Task> loadData() throws JarvisException {
+        try {
+            List<Task> lst = new ArrayList<>();
+            File dataFile = new File(dataFilePath);
+            if (dataFile.exists()) {
+                Scanner s = new Scanner(dataFile);
+                while (s.hasNext()) {
+                    String[] line = s.nextLine().split("\\|");
+                    Task task;
```

Comment:
Maybe you might want to declare the task variable inside the switch cases? 
Something like `Task task = new Todo(line[2]);`